openmmtools/tests/test_integrators.py

```diff
@@ -310,6 +310,105 @@ def test_external_protocol_work_accumulation():
     del context, integrator
 
 class TestExternalPerturbationLangevinIntegrator(TestCase):
+
+    def create_system(self, testsystem, parameter_name, parameter_initial, temperature = 298.0 * unit.kelvin, platform_name='Reference'):
+        """
+        Create an example system to be used by other tests
+        """
+        system, topology = testsystem.system, testsystem.topology
+        integrator = integrators.ExternalPerturbationLangevinIntegrator(splitting="O V R V O", temperature=temperature)
+
+        # Create the context
+        platform = openmm.Platform.getPlatformByName(platform_name)
+        if platform_name in ['CPU', 'CUDA']:
+            platform.setPropertyDefaultValue('DeterministicForces', 'true')
+        context = openmm.Context(system, integrator, platform)
+        context.setParameter(parameter_name, parameter_initial)
+        context.setPositions(testsystem.positions)
+        context.setVelocitiesToTemperature(temperature)
+
+        return context, integrator
+
+    def run_ncmc(self, context, integrator, temperature, nsteps, parameter_name, parameter_initial, parameter_final):
+        """
+        A simple example of NCMC to be used with unit tests. The protocol work should be reset each time this command
+        is called.
+
+        Returns
+        -------
+        external_protocol_work: float
+            the protocol work calculated with context.getState()
+        integrator_protocol_work: float
+            the protocol work calculated inside the integrator.
+        """
+
+        kT = kB * temperature
+
+        external_protocol_work = 0.0
+        integrator.step(1)
+        for step in range(nsteps):
+            lambda_value = float(step + 1) / float(nsteps)
+            parameter_value = parameter_initial * (1 - lambda_value) + parameter_final * lambda_value
+            initial_energy = context.getState(getEnergy=True).getPotentialEnergy()
+            context.setParameter(parameter_name, parameter_value)
+            final_energy = context.getState(getEnergy=True).getPotentialEnergy()
+            external_protocol_work += (final_energy - initial_energy) / kT
+            integrator.step(1)
+
+        integrator_protocol_work = integrator.getGlobalVariableByName('protocol_work') * unit.kilojoules_per_mole / kT
```

Comment:
Don't use the global variables to access work! Always use `integrator.get_protocol_work(dimensionless=True)` if you want the dimensionless work!